tests/integration-tests/src/commonTest/kotlin/test/WatcherTest.kt

```diff
@@ -74,19 +81,26 @@ class WatcherTest {
   fun sameQueryTriggersWatcher() = runTest(before = { setUp() }) {
     val query = EpisodeHeroNameQuery(Episode.EMPIRE)
     val channel = Channel<EpisodeHeroNameQuery.Data?>()
+    val channel2 = Channel<EpisodeHeroNameQuery.Data?>()
```

Comment:
Replacing these 2 lines with 
```kotlin
    val channel = Channel<EpisodeHeroNameQuery.Data?>(Channel.UNLIMITED)
    val channel2 = Channel<EpisodeHeroNameQuery.Data?>(Channel.UNLIMITED)
```

also fixes the test for me. 

I must admit I don't understand why. Without it, the `send` could block?

Reply:
Ohhh interesting, maybe the Channels all share the same worker underneath. Not sure why it would block though yeah 🤔

Reply:
The test passes in our CI as well with this change 🤔 
I'm not sure what could have happened. Maybe a recent change affected the watchers scopes underneath?

Reply:
Channel.BUFFERED works as well. Wondering if the first `receiveOrTimeout()` is not somehow affecting the other Channel as well 🤔  although it wasn't before.